app/src/test/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegrationTests.java

```diff
@@ -0,0 +1,271 @@
+        AppsflyerIntegration.setManualMode(false);
+        Assert.assertFalse(AppsflyerIntegration.manualMode);
+    }
+
+    @Test
+    public void testAppsflyerIntegration_startAppsFlyer_happyFlow() throws Exception {
+        MockedStatic<AppsFlyerLib> staticAppsFlyerLib = mockStatic(AppsFlyerLib.class);
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        staticAppsFlyerLib.when(AppsFlyerLib::getInstance).thenReturn(appsFlyerLib);
+        Context context = mock(Context.class);
+
+        AppsflyerIntegration.startAppsFlyer(context);
+
+        verify(appsFlyerLib).start(context);
+
+        reset(appsFlyerLib,context);
+        staticAppsFlyerLib.close();
+    }
+
+    @Test
+    public void testAppsflyerIntegration_startAppsFlyer_nilFlow() throws Exception {
+        MockedStatic<AppsFlyerLib> staticAppsFlyerLib = mockStatic(AppsFlyerLib.class);
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        staticAppsFlyerLib.when(AppsFlyerLib::getInstance).thenReturn(appsFlyerLib);
+
+        AppsflyerIntegration.startAppsFlyer(null);
+
+        verify(appsFlyerLib,never()).start(any());
+
+        reset(appsFlyerLib);
+        staticAppsFlyerLib.close();
+    }
+
+    @Test
+    public void testAppsflyerIntegration_FACTORYCreate_happyFlow() throws Exception {
+        MockedStatic<AppsFlyerLib> staticAppsFlyerLib = mockStatic(AppsFlyerLib.class);
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        staticAppsFlyerLib.when(AppsFlyerLib::getInstance).thenReturn(appsFlyerLib);
+        Analytics analytics = mock(Analytics.class);
+        ValueMap settings = new ValueMap();
+        settings.put("appsFlyerDevKey" , "devKey");
+        settings.put("trackAttributionData" , true);
+        Logger logger = new Logger("test", Analytics.LogLevel.INFO);
+        Mockito.when(analytics.logger("AppsFlyer")).thenReturn(logger);
+        Application app = mock(Application.class);
+        Mockito.when(analytics.getApplication()).thenReturn(app);
+        AppsflyerIntegration.deepLinkListener = mock(AppsflyerIntegration.ExternalDeepLinkListener.class);
+
+        Integration<AppsFlyerLib> integration= (Integration<AppsFlyerLib>) AppsflyerIntegration.FACTORY.create(settings,analytics);
+
+        verify(appsFlyerLib).setDebugLog(logger.logLevel!=Analytics.LogLevel.NONE);
+        ArgumentCaptor<AppsflyerIntegration.ConversionListener> captorListener = ArgumentCaptor.forClass(AppsflyerIntegration.ConversionListener.class);
+        ArgumentCaptor<String> captorDevKey = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Context> captorContext = ArgumentCaptor.forClass(Context.class);
+        verify(appsFlyerLib).init(captorDevKey.capture(), captorListener.capture() , captorContext.capture());
+        Assert.assertNotEquals(captorListener.getValue(), null);
+        Assert.assertTrue(captorListener.getValue() instanceof AppsflyerIntegration.ConversionListener);
```

Comment:
Condition `captorListener.getValue() instanceof AppsflyerIntegration.ConversionListener` is redundant and can be replaced with a null check which is what you did in the previous line.